### PR TITLE
Update copy on various qualification pages

### PR DESCRIPTION
**Pull request by JonF1**
## Context
Copy changes needed for subject pages

## Changes proposed in this pull request
* Simplify yml files and refactor `gcse_grade_guidance_component` erb and specs
* Remove helper method + refactor partial translations
* Fix feature specs
## Guidance to review
Verify new copy is the same as the google doc in trello card

## Link to Trello card

https://trello.com/c/gZPiZotO/4293-update-copy-on-various-qualification-pages

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb [hunk 1]
```diff
@@ -1,139 +1,80 @@
...
-      expect(result.text).to include(t('gcse_edit_grade.guidance.main'))
-    end
-
-    it 'does not display the guidance around english literature and multiple english qualifications' do
-      subject = 'science'
-
-      result = render_inline(described_class.new(subject, nil))
+    it 'displays the O level guidance' do
+      subjects.each do |s|
+        result = render_inline(described_class.new(s, qualification_type))
```
Comment (despo) [inline spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb:11]: Can the specs be updated to use meaninful variable names e.g.

```suggestion
      subjects.each do |subject|
        result = render_inline(described_class.new(subject, qualification_type))
```

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb [hunk 2]
```diff
@@ -1,139 +1,80 @@
...
-      it 'displays the guidance around triple GCSE science' do
-        subject = 'science'
-        qualification_type = 'gce_o_level'
+  context 'when the qualification type is GCSE for any subject' do
+    let(:qualification_type) { 'gcse' }
 
-        result = render_inline(described_class.new(subject, qualification_type))
+    it 'displays the GCSE guidance' do
+      subjects.each do |s|
+        result = render_inline(described_class.new(s, qualification_type))
```
Comment (despo) [inline spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb:23]: same as above

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb [hunk 3]
```diff
@@ -1,139 +1,80 @@
...
-      it 'does not display the guidance around triple science or three science subjects' do
-        subject = 'science'
-        qualification_type = 'other_uk'
-
-        result = render_inline(described_class.new(subject, qualification_type))
+    context 'when the subject is Maths or Science' do
+      it 'does not show an additional guidance message' do
+        subjects = %w[maths science]
+        subjects.each do |s|
+          result = render_inline(described_class.new(s, qualification_type))
```
Comment (despo) [inline spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb:42]: same as above

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb [hunk 4]
```diff
@@ -1,139 +1,80 @@
...
-  context 'when the subject is english' do
-    it 'displays the guidance around the expectation of providers for multiple English GCSEs' do
-      subject = 'english'
+  context 'when the qualification type is Scottish National 5 for any subject' do
+    let(:qualification_type) { 'scottish_national_5' }
 
-      result = render_inline(described_class.new(subject, 'gcse'))
+    it 'displays the Scottish National 5 guidance' do
+      subjects.each do |s|
+        result = render_inline(described_class.new(s, qualification_type))
```
Comment (despo) [inline spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb:55]: same as above

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb [hunk 5]
```diff
@@ -1,139 +1,80 @@
...
-      it 'does not display the guidance around triple science or three science subjects' do
-        subject = 'english'
-        qualification_type = 'other_uk'
-
-        result = render_inline(described_class.new(subject, qualification_type))
+    context 'when the subject is Maths or English' do
+      it 'does not show an additional guidance message' do
+        subjects = %w[maths english]
+        subjects.each do |s|
+          result = render_inline(described_class.new(s, qualification_type))
```
Comment (despo) [inline spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb:74]: same as above

File: app/components/candidate_interface/gcse_grade_guidance_component.html.erb [hunk 6]
```diff
@@ -1,28 +1,15 @@
-<% unless @subject == 'english' %>
   <p class="govuk-body">
```
Comment (despo) [inline app/components/candidate_interface/gcse_grade_guidance_component.html.erb:1]:
```suggestion
<p class="govuk-body">
```

File: app/components/candidate_interface/gcse_grade_guidance_component.html.erb [hunk 7]
```diff
@@ -1,28 +1,15 @@
...
+  <% elsif scottish_national_5? %>
+    <%= simple_format t('gcse_edit_grade.guidance.scottish_national_5'), class: 'govuk-body' %>
+  <% else %>
+    <%= simple_format t('gcse_edit_grade.guidance.main'), class: 'govuk-body' %>
+    <% if english? %>
+      <%= tag.p(t('gcse_edit_grade.guidance.multiple_english_gcses.secondary'), class: 'govuk-body') %>
     <% end %>
-  </p>
-<% end %>
+  <% end %>
```
Comment (despo) [inline app/components/candidate_interface/gcse_grade_guidance_component.html.erb:11]:
```suggestion
  <% end %>
</p>
```

File: app/components/candidate_interface/gcse_grade_guidance_component.html.erb [hunk 8]
```diff
@@ -1,28 +1,15 @@
...
-    <% else %>
-      <%= simple_format t('gcse_edit_grade.guidance.main'), class: 'govuk-body' %>
+  <% if gce_o_level? %>
+    <%= simple_format t('gcse_edit_grade.guidance.o_level_guidance'), class: 'govuk-body' %>
+  <% elsif scottish_national_5? %>
+    <%= simple_format t('gcse_edit_grade.guidance.scottish_national_5'), class: 'govuk-body' %>
+  <% else %>
+    <%= simple_format t('gcse_edit_grade.guidance.main'), class: 'govuk-body' %>
+    <% if english? %>
+      <%= tag.p(t('gcse_edit_grade.guidance.multiple_english_gcses.secondary'), class: 'govuk-body') %>
```
Comment (despo) [inline app/components/candidate_interface/gcse_grade_guidance_component.html.erb:9]: To keep things consistent it's best to stick to using the html tag format as it makes it easier to figure out what the output is

```suggestion
     <p class="govuk-body">
        <%= t('gcse_edit_grade.guidance.multiple_english_gcses.secondary') %>
     </p>
```

File: app/components/candidate_interface/gcse_grade_guidance_component.html.erb [hunk 9]
```diff
@@ -1,28 +1,15 @@
...
-
-<% if english? && (gcse? || gce_o_level?) %>
-  <p class="govuk-body">
-    <%= t('gcse_edit_grade.guidance.multiple_english_gcses.main') %>
-  </p>
-  <p class="govuk-body">
-    <%= t('gcse_edit_grade.guidance.multiple_english_gcses.secondary') %>
-  </p>
+<% if science? && scottish_national_5? %>
+    <%= tag.p(t('gcse_edit_grade.guidance.triple_scottish_national_science'), class: 'govuk_body') %>
```
Comment (despo) [inline app/components/candidate_interface/gcse_grade_guidance_component.html.erb:14]:
```suggestion
    <p class="govuk_body">
         <%= t('gcse_edit_grade.guidance.triple_scottish_national_science') %>
    </p>
```

File: spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb (hunk 1 above)
Comment (JonF1) [reply spec/components/candidate_interface/gcse_grade_guidance_component_spec.rb]: Thanks, fixed

**Review by frankieroberto — CHANGES_REQUESTED** [review]

**Comment by EmmaFrith** [comment]
Sorry @JonF1, this wasn't covered in my copy change requests, but I just spotted the following...

When selecting 'Other UK qualification'

Screenshot 2022-01-17 at 12 46 23

The candidate sees this page:

Screenshot 2022-01-17 at 12 49 05

In this scenario, could we change the copy to the following:

Screenshot 2022-01-17 at 12 52 56

File: config/locales/candidate_interface/gcse_details.yml [hunk 10]
```diff
@@ -19,35 +19,35 @@ en:
     page_title: What grade is your %{subject} %{qualification_type}?
     guidance:
       main: |
-        Providers expect you to have a grade C or 4 (or the equivalent) in order
-        to train.
+        You need at least a grade 4 (C).
 
-        If you do not have this, contact your provider to discuss your
-        options. You might be able to retake an exam or do an equivalence test.
+        If you do not have this, you can contact teacher training providers to discuss your eligibility for their courses. You might be able to retake an exam or do an equivalence test.
```
Comment (EmmaFrith) [inline config/locales/candidate_interface/gcse_details.yml:24]:
```suggestion
        If you do not have this, contact teacher training providers to discuss your eligibility for their courses. You might be able to retake an exam or do an equivalence test.
```

File: config/locales/candidate_interface/gcse_details.yml [hunk 11]
```diff
@@ -19,35 +19,35 @@ en:
...
       o_level_guidance: |
-        Providers expect you to have a grade C or above in order to train.
+        You need at least a grade C or equivalent.
 
-        If you do not have this, contact your provider to discuss your
-        options. You might be able to retake an exam or do an equivalence test.
-      o_level_triple_gcse_science: If you have a combined or triple O Level in science, enter your total grade.
+        If you do not have this, you can contact teacher training providers to discuss
```
Comment (EmmaFrith) [inline config/locales/candidate_interface/gcse_details.yml:28]:
```suggestion
        If you do not have this, contact teacher training providers to discuss
```

File: config/locales/candidate_interface/gcse_details.yml [hunk 12]
```diff
@@ -19,35 +19,35 @@ en:
...
+        your eligibility for their courses. You might be able to do an equivalence test.
+      scottish_national_5: |
+        You need at least a grade C.
+
+        If you do not have this, you can contact teacher training providers to discuss
```
Comment (EmmaFrith) [inline config/locales/candidate_interface/gcse_details.yml:33]:
```suggestion
        If you do not have this, contact teacher training providers to discuss
```

File: config/locales/candidate_interface/gcse.yml [hunk 13]
```diff
@@ -20,7 +20,7 @@ en:
       not_completed_explanation:
         label: If you are working towards this qualification, give us details (optional)
       grade:
-        label: Please specify your grade
+        label: Grade
       award_year:
         label: Enter year
```
Comment (EmmaFrith) [inline config/locales/candidate_interface/gcse.yml:25]:
```suggestion
        label: Year
```

File: app/components/candidate_interface/gcse_grade_guidance_component.html.erb [hunk 14]
```diff
@@ -1,28 +1,20 @@
...
-  </p>
-<% end %>
-
-<% if science? %>
-  <p class="govuk-body">
-    <% if scottish_national_5? %>
-      <%= t('gcse_edit_grade.guidance.triple_scottish_national_science') %>
-    <% elsif gcse? || gce_o_level? %>
-      <%= t('gcse_edit_grade.guidance.o_level_triple_gcse_science') %>
+<p class="govuk-body">
```
[the remark this answers is not in the document]
Comment (JonF1) [reply app/components/candidate_interface/gcse_grade_guidance_component.html.erb]: Thanks, fixed

File: config/locales/candidate_interface/gcse_details.yml (hunk 10 above)
Comment (JonF1) [reply config/locales/candidate_interface/gcse_details.yml]: Thanks, fixed

**Comment by JonF1** [comment]
> Sorry @JonF1, this wasn't covered in my copy change requests, but I just spotted the following...
> 
> When selecting 'Other UK qualification' here:
> 
> Screenshot 2022-01-17 at 12 46 23
> 
> The candidate sees this page:
> 
> Screenshot 2022-01-17 at 12 49 05
> 
> In this scenario, could we change the copy to the following:
> 
> Screenshot 2022-01-17 at 12 52 56
> 
> This would apply for not only the English section but the Maths and Science sections too

Good spot! Thanks, I've pushed up the changes

**Review by EmmaFrith — APPROVED** [review]
Thanks so much @JonF1! LGTM from a content p.o.v :)

File: config/locales/candidate_interface/gcse_details.yml [hunk 15]
```diff
@@ -19,35 +19,40 @@ en:
...
-        options. You might be able to retake an exam or do an equivalence test.
+        If you do not have this, contact teacher training providers to discuss your eligibility for their courses. You might be able to retake an exam or do an equivalence test.
       o_level_guidance: |
-        Providers expect you to have a grade C or above in order to train.
+        You need at least a grade C or equivalent.
 
-        If you do not have this, contact your provider to discuss your
-        options. You might be able to retake an exam or do an equivalence test.
-      o_level_triple_gcse_science: If you have a combined or triple O Level in science, enter your total grade.
+        If you do not have this, contact teacher training providers to discuss
```
Comment (stevehook) [inline config/locales/candidate_interface/gcse_details.yml:28]: I think `simple_format` is inserting `<br>` for the line breaks here...

image

**Review by stevehook — COMMENTED** [review]
This all looks good to merge except for a couple of a few line breaks in the .yml file that don't render as we want.

File: config/locales/candidate_interface/gcse_details.yml (hunk 15 above)
Comment (JonF1) [reply config/locales/candidate_interface/gcse_details.yml]: thanks, fixed

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**